extension/healthcheckextension/healthcheckextension_test.go

```diff
@@ -86,14 +85,8 @@ func TestHealthCheckExtensionPortAlreadyInUse(t *testing.T) {
 	hcExt := newServer(config, zap.NewNop())
 	require.NotNil(t, hcExt)
 
-	// Health check will report port already in use in a goroutine, use the error waiting
-	// host to get it.
-	mh := componenttest.NewErrorWaitingHost()
-	require.NoError(t, hcExt.Start(context.Background(), mh))
-
-	receivedError, receivedErr := mh.WaitForFatalError(500 * time.Millisecond)
-	require.True(t, receivedError)
-	require.Error(t, receivedErr)
+	mh := componenttest.NewAssertNoError(t)
+	require.Error(t, hcExt.Start(context.Background(), mh))
```

Comment:
This seems to be flaky because it is not guaranteed that this test does not end before the fatal error is reported.

Reply:
The previous code was incorrectly using ReportFatalError in this case and returning nil in case of port already in use. I changed the code to directly return the error in this case. See https://github.com/open-telemetry/opentelemetry-collector/pull/2679/files#diff-181eaa0bc7b98757e011c0f7a51d93566cc38f60cb9419278631fc6cf9c87bdcL44